docs/notebooks/inequality_constraints.pct.py

```diff
@@ -161,6 +169,127 @@ def create_bo_model(data):
 )
 plt.show()
 
+# %% [markdown]
+# Now, we will show how to solve the same problem, but when point are queried in (synchronous) batches instead of one at a time.
+
+# %%
+
+
+class BatchExpectedConstrainedImprovement(BatchAcquisitionFunctionBuilder):
+    """
+    Builder for the _expected constrained improvement_ acquisition function defined in
+    Gardner, 2014, in a batch form, relying on Monte-Carlo.
+    """
+    def __init__(
+        self,
+        objective_tag: str,
+        constraint_builder: AcquisitionFunctionBuilder,
+        num_samples,
+        num_query_points,
+        threshold,
+        min_feasibility_probability: Union[float, tf.Tensor] = 0.5,
+    ):
+
+        tf.debugging.assert_scalar(min_feasibility_probability)
+
+        if not 0 <= min_feasibility_probability <= 1:
+            raise ValueError(
+                f"Minimum feasibility probability must be between 0 and 1 inclusive,"
+                f" got {min_feasibility_probability}"
+            )
+
+        self._objective_tag = objective_tag
+        self._constraint_builder = constraint_builder
+        self._min_feasibility_probability = min_feasibility_probability
+        self.num_samples = num_samples
+        self.num_query_points = num_query_points
+        self.threshold = threshold
+
+    def prepare_acquisition_function(
+        self, datasets: Mapping[str, Dataset], models: Mapping[str, ProbabilisticModel]
+    ) -> AcquisitionFunction:
+        objective_model = models[self._objective_tag]
+        objective_dataset = datasets[self._objective_tag]
+
+        if len(objective_dataset) == 0:
+            raise ValueError(
+                "Expected improvement is defined with respect to existing points in the objective"
+                " data, but the objective data is empty."
+            )
+
+        constraint_fn = self._constraint_builder.prepare_acquisition_function(datasets, models)
+        pof = constraint_fn(objective_dataset.query_points)
+        is_feasible = pof >= self._min_feasibility_probability
+
+        if not tf.reduce_any(is_feasible):
+            return constraint_fn
+
+        mean, _ = objective_model.predict(objective_dataset.query_points)
+        eta = tf.reduce_min(tf.boolean_mask(mean, is_feasible), axis=0)
+
+        eps = {tag:
```

Comment:
writing this loop here is probably something we want to avoid. Can we find a way to include it in the `sampler`?

Reply:
seems like the same as https://github.com/secondmind-labs/trieste/pull/93#discussion_r536160461